tracing-subscriber/src/filter/subscriber_filters/combinator.rs

```diff
@@ -421,6 +431,14 @@ where
         None
     }
 
+    #[inline]
+    fn event_enabled(&self, event: &tracing_core::Event<'_>, cx: &Context<'_, S>) -> bool {
+        // Never disable based on event_enabled; we "disabled" it in `enabled`,
+        // so the `not` has already been applied and filtered this not out.
```

Comment:
hmm...i wonder if it's worth having a separate filter combinator specifically for inverting the value of `event_enabled`, although that seems _extremely_ niche. regardless, i think it would be nice to explicitly mention `event_enabled` in the top level docs for the `not` combinator...

Reply:
Agreed on both points that

- inverting `event_enabled` is extremely niche
  - inverting `enabled && event_enabled` (`(!enabled) || (!event_enabled)`) is at least somewhat involved
    - the case of `enabled = false, event_enabled = true` is just cursed to think about, but we deliberately don't require consistency here
      - I *think* we properly documented that `event_enabled` can't override a previous `enabled = false` sufficiently
      - but unless you effortlessly know DeMorgan's, reasoning about the inversion is not trivial
  - but 99.9% of filters not in tracing-filter won't implement `event_enabled`
  - and properly implementing `(!enabled) || (!event_enabled)` requires extra state to remember the result of `enabled` (or "just" calling `enabled` late[^1], I suppose)
- the docs should mention only `enabled` is called by `not`

[^1]:
    ```rust
    fn enabled(&self, meta: &Metadata<'_>, cx: &Context<'_, S>) -> bool {
        !(meta.is_span() && self.a.enabled(meta, cx))
    }
    fn event_enabled(&self, event: &tracing_core::Event<'_>, cx: &Context<'_, S>) -> bool {
        !(self.a.enabled(event.metadata(), cx) && self.a.event_enabled(event, cx)
    }
    ```
    but this would prevent the all-layers-individually-said-no optimization on `enabled` for events.